util.cc

```diff
@@ -258,6 +258,23 @@ xfopen(const char* fname, const char* type, const char* errtxt)
   return f;
 }
 
+/*
+ * Thin wrapper around fopen() that supports UTF-8 fname on all platforms.
+ */
+FILE*
+ufopen(const char* fname, const char* mode)
+{
+#if __WIN32__
+  // On Windows standard fopen() doesn't support UTF-8, so we have to convert
+  // to wchar_t* (UTF-16) and use the wide-char version of fopen(), _wfopen().
+  return _wfopen((const wchar_t*) QString(fname).utf16(),
+                 (const wchar_t*) QString(mode).utf16());
+#else
+  // On other platforms, convert to native locale (UTF-8 or other 8-bit).
+  return fopen(qPrintable(QString(fname)), mode);
+#endif
+}
+
```

Comment:
I think it would be better to use one method for QString -> wchar_t conversion.  gbfile is now using a different method that util.cc.  Let's settle on one.

Reply:
Done.
kml.cc and util.cc are using the QString::utf16() to const wchar_t* cast, so I've updated gbfile.cc to use that too.